core/domain/topic_services.py

```diff
@@ -261,9 +329,13 @@ def apply_change_list(topic_id, change_list):
     try:
         for change in change_list:
             if change.cmd == topic_domain.CMD_ADD_SUBTOPIC:
-                topic.add_subtopic(change.id, change.title)
+                subtopic_page = topic.add_subtopic(change.title)
+                save_subtopic_page_model(subtopic_page)
```

Comment:
I don't think this works, what if applying later parts of the changelist results in an error? Then the whole thing gets rolled back but the subtopic page model is still in the datastore.

So, may need to move the subtopic page logic higher up so that all the models are created at the same time.

Might want to look at how we currently add explorations to collections, the flow seems like it would be similar (I mean the one where we create a new exploration and add its ID to the collection).

Reply:
In adding explorations to collections, it is added as soon as the button is clicked in the frontend, and then appended to the collection.

Reply:
Done, I retrieved the new subtopic_id here and created the subtopic page after the update for the topic is complete.